testgrid/cmd/transfigure/cmd/transfigure.go

```diff
@@ -0,0 +1,363 @@
+/*
+Copyright © 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package transfigure
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/spf13/cobra"
+)
+
+const Branch = "transfigure-branch"
+
+var (
+	dryRun         bool
+	gitEmail       string
+	githubToken    string
+	gitUser        string
+	prowConfig     string
+	prowJobConfig  string
+	prTitle        string
+	remoteForkRepo string
+	repoSubdir     string
+	tokenContents  string
+	testgridDir    string
+	testgridYaml   string
+	workingDir     string
+)
+
+// rootCmd represents the base command when called without any subcommands
+var rootCmd = &cobra.Command{
+	Use:   "transfigure",
+	Short: "Generates a YAML Testgrid config from a Prow config and pushes it to testgrid.k8s.io.",
+	Long: `Transfigure is an image that generates a YAML Testgrid configuration
+		   from a Prow configuration and pushes it to be used on testgrid.k8s.io. It 
+	       is used specifically for Prow instances other than the k8s instance of Prow.`,
+	Run: func(cmd *cobra.Command, args []string) {
+		err := run()
+		if err != nil {
+			log.Fatal(err)
+		}
+	},
+}
+
+// Execute adds all child commands to the root command and sets flags appropriately.
+// This is called by main.main(). It only needs to happen once to the rootCmd.
+func Execute() {
+	if err := rootCmd.Execute(); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+}
+
+func run() error {
+	if !dryRun && githubToken == "" {
+		return errors.New("Run was not a dry run, and flag 'github_token' was not set.")
+	}
+
+	err := createTempWorkingDir()
+	if err != nil {
+		return err
+	}
```

Comment:
The Go way:
```suggestion
	if err := createTempWorkingDir(); err != nil {
		return err
	}
```

And same below

Reply:
Done.